test/src/test/java/org/corfudb/integration/StreamingIT.java

```diff
@@ -323,4 +321,67 @@ public void testStreamingMultiTableStreams() throws Exception {
 
         assertThat(shutdownCorfuServer(corfuServer)).isTrue();
     }
+
+    /**
+     * Streaming Test with 2 different tables and a single streamer
+     * <p>
+     * The test creates two tables in the same namespace and makes updates to both in the same transaction.
+     * A single streamer subscribes to updates from both.
+     * <p>
+     * The test verifies that the streamer receives updates from both tables and since the updates were made in
+     * the same transaction, they are received in one CorfuStreamEntry.
+     * @throws Exception
+     */
+    @Test
+    public void testStreamingMultiTableSingleListener() throws Exception {
+        // Run a corfu server
+        Process corfuServer = runSinglePersistentServer(corfuSingleNodeHost, corfuStringNodePort);
+
+        // Start a Corfu runtime
+        runtime = createRuntime(singleNodeEndpoint);
+
+        runtime.setTransactionLogging(true);
+        CorfuStore store = new CorfuStore(runtime);
+
+        // Record the initial timestamp.
+        Timestamp ts1 = store.getTimestamp();
+
+        // Create 2 tables in the same namespace
+        Table<Uuid, Uuid, Uuid> n1t1 = store.openTable(
+                "n1", "t1", Uuid.class,
+                Uuid.class, Uuid.class,
+                TableOptions.builder().build()
+        );
+
+        Table<Uuid, Uuid, Uuid> n2t1 = store.openTable(
+                "n1", "t2", Uuid.class,
+                Uuid.class, Uuid.class,
+                TableOptions.builder().build()
+        );
+
+        // Make an update to the tables in a transaction
+        final int t1_uuid = 5;
+        final int t2_uuid = 10;
+        Uuid t1Uuid = Uuid.newBuilder().setMsb(t1_uuid).setLsb(t1_uuid).build();
+        Uuid t2Uuid = Uuid.newBuilder().setMsb(t2_uuid).setLsb(t2_uuid).build();
+        TxBuilder txBuilder = store.tx("n1");
+        txBuilder.update("t1", t1Uuid, t1Uuid, t1Uuid);
+        txBuilder.update("t2", t2Uuid, t2Uuid, t2Uuid);
+        txBuilder.commit();
+
+        // Subscribe to both tables
+        List<TableSchema> tablesSubscribed = new ArrayList<>();
+        tablesSubscribed.add(new TableSchema("t1", Uuid.class, Uuid.class, Uuid.class));
+        tablesSubscribed.add(new TableSchema("t2", Uuid.class, Uuid.class, Uuid.class));
+        StreamListenerImpl listener = new StreamListenerImpl("n1_listener");
+        store.subscribe(listener, "n1", tablesSubscribed, ts1);
+
+        // Verify that both updates come to the subscriber in the same StreamEntry
+        TimeUnit.SECONDS.sleep(2);
```

Comment:
Why sleeping here? Is the test relying on the sleep to be passed? If not we can remove it, otherwise it's better to make it synchronous instead of sleeping.

Reply:
there is an async thread which delivers the updates every 50ms.  Changing it to be synchronous is a bigger change so keeping the sleep for now.